python/tests/test_adjoint_solver.py

```diff
@@ -505,21 +505,21 @@ def test_complex_fields(self):
             ## compute gradient using adjoint solver
             adjsol_obj, adjsol_grad = adjoint_solver_complex_fields(p, frequencies)
 
-            ## compute unperturbed |Ez|^2
-            Ez2_unperturbed = forward_simulation_complex_fields(p, frequencies)
+            ## compute unperturbed |Dz|^2
+            Dz2_unperturbed = forward_simulation_complex_fields(p, frequencies)
 
             ## compare objective results
-            print("Ez2 -- adjoint solver: {}, traditional simulation: {}".format(adjsol_obj,Ez2_unperturbed))
-            self.assertClose(adjsol_obj,Ez2_unperturbed,epsilon=1e-6)
+            print("Dz2 -- adjoint solver: {}, traditional simulation: {}".format(adjsol_obj,Dz2_unperturbed))
+            self.assertClose(adjsol_obj,Dz2_unperturbed,epsilon=1e-6)
```

Comment:
Looks like this assert statement is failing for the single-precision build. You can try increasing the tolerance for single precision using e.g.:
```
self.assertClose(adjsol_obj,Dz2_unperturbed,epsilon=1e-4 if mp.is_single_precision() else 1e-6)
```

Reply:
In my place, this line does not cause error. I guess the error message in my place is different from that given by the online check. But I cannot see the detailed error message in the online-check logs. Can I ask where I should find the detailed error message?

Reply:
To find the log of the failing CI run, click on the "Details" button below next to "CI / Test Python 3.10 with MPI (true)" or the "Actions" tab above (next to "Discussions" and "Projects"). From there, for the failing run locate the "Artifacts" section and then download the log file. In this case, this would be [py3.10-tests-mpi-true.log](https://github.com/NanoComp/meep/actions/runs/2513272770) which shows:
```
Ran 7 tests in 307.004s

FAILED (failures=1)
FAIL: test_complex_fields (__main__.TestAdjointSolver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/meep/meep/build/meep-1.24.0-beta/_build/sub/python/../../../python/tests/test_adjoint_solver.py", line 525, in test_complex_fields
    self.assertClose(adj_scale,fd_grad,epsilon=tol)
  File "/home/runner/work/meep/meep/build/meep-1.24.0-beta/python/tests/utils.py", line 28, in assertClose
    self.assertLessEqual(diff_norm, epsilon * np.maximum(x_norm, y_norm), msg)
AssertionError: 0.001776412917294544 not less than or equal to 0.0005871758943843297 : 
```

Reply:
Thanks! The tolerance for single precision is adjusted further.